src/SearchTrips/Trip.tsx

```diff
@@ -0,0 +1,156 @@
+import { Fragment } from "react";
+import { Feature, Leg, TripPattern } from "../types";
+import { ActionPanel, Action, Color, List } from "@raycast/api";
+import { fetchTrip } from "../api";
+import { useCallback, useEffect, useState } from "react";
+import {
+  getTransportIcon,
+  formatAsClock,
+  formatTimeDifferenceAsClock,
+  formatMetersToHuman,
+} from "../utils";
+import Accessory = List.Item.Accessory;
+
+const buildAccessories = (trip: TripPattern, isDetailsVisible: boolean): Accessory[] => {
+  if (isDetailsVisible) {
+    return [
+      {
+        tag: {
+          value: formatTimeDifferenceAsClock(trip.expectedStartTime, trip.expectedEndTime),
+          color: Color.Blue,
+        },
+      },
+    ];
+  }
+  const accessories: Accessory[] = trip.legs
+    .filter((leg) => !(leg.mode === "foot" && leg.fromPlace.name === leg.toPlace.name))
+    .map((leg) => ({
+      icon: getTransportIcon(leg.mode, leg.transportSubmode),
+      text: `${leg.line?.publicCode || ""}`,
```

Comment:
We can pass `undefined` or `null` here, so I don't think we should fall back to the empty string

```suggestion
      text: leg.line?.publicCode,
```